msdk-rawdata/msdk-rawdata-peakinvestigator/src/main/java/io/github/msdk/rawdata/peakinvestigator/PeakInvestigatorMsSpectrum.java

```diff
@@ -0,0 +1,125 @@
+/* 
+ * (C) Copyright 2015-2016 by MSDK Development Team
+ *
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+
+package io.github.msdk.rawdata.peakinvestigator;
+
+import java.util.TreeMap;
+
+import com.google.common.collect.Range;
+
+import io.github.msdk.datamodel.msspectra.MsSpectrum;
+import io.github.msdk.datamodel.msspectra.MsSpectrumType;
+import io.github.msdk.util.tolerances.MzTolerance;
+
+/**
+ * This class is used provide m/z-dependent mass tolerances from
+ * PeakInvestigator by decorating an existing instance of MsSpectrum.
+ *
+ */
+public class PeakInvestigatorMsSpectrum implements MsSpectrum, MzTolerance {
```

Comment:
For example, in MZmine we use a shoulder peaks filter for FTMS data, to remove residual signals from the Fourier Transform. It's not implemented in MSDK yet, just a stub (https://github.com/msdk/msdk/tree/master/msdk-spectra/msdk-spectra-shoulderpeaks).

Yes, the Chromatogram Builder can have a default mzTolerance in case the spectrum has null.